### PR TITLE
[NameLookup] Fix opaque decl astscope

### PR DESCRIPTION
**Pull request by davidungar**
Do not try to do lookups on OpaqueDecls that are in inactive clauses.
Also, fail the compilation for any lookups into inactive clauses.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
@swift-ci please test

File: lib/AST/ASTScopeLookup.cpp [hunk 1]
```diff
@@ -82,6 +84,12 @@ const ASTScopeImpl *ASTScopeImpl::findStartingScopeForLookup(
   if (!startingScope) {
     llvm::errs() << "ASTScopeImpl: resorting to startingScope hack, file: "
                  << sourceFile->getFilename() << "\n";
+    // The check is costly, and inactive lookups will end up here, so don't
+    // do the check unless we can't find the startingScope.
+    const bool isInInactiveClause =
```
Comment (DougGregor) [inline lib/AST/ASTScopeLookup.cpp]: I don’t understand why we need this part, of the parser isn’t going to add these decls in the first place for inactive regions?

Comment (davidungar) [reply lib/AST/ASTScopeLookup.cpp]: The idea is that lookups to things in inactive clauses may creep back in at some point. If they do, this part will output a more specific message for debugging, and also, as per your preference, will crash the compilation. It only runs when the ASTScope code cannot find the starting scope anyway, so it doesn't slow down compilation unless there's an error.

When someday, we rip out all the code that passed DeclContexts into UnqualifiedLookup, the whole did-not-find starting scope stuff will get ripped out.

Comment (DougGregor) [reply lib/AST/ASTScopeLookup.cpp]: Okay!

**Comment by davidungar** [comment]
As per offline conversation with @DougGregor , am proceeding to merge.

**Comment by davidungar** [comment]
Likely fixes rdar://56334618

**Pull request merged**